Assets/Scripts/ConnectionManagement/ConnectionState/ClientReconnectingState.cs

```diff
@@ -21,6 +21,9 @@ class ClientReconnectingState : ClientConnectingState
         string m_LobbyCode = "";
         int m_NbAttempts;
 
+        const float k_TimeBetweenLobbyReconnectAttempts = 1;
+        const float k_NbLobbyReconnectAttempts = 5;
```

Comment:
any reason these are different from UTP reconnect attempts?

Reply:
I could use the parameters we have set to our instance of UTP instead of these constants, but that wouldn't make sense in my opinion, since they don't serve the same purpose

Reply:
I was talking about ConnectionManager's m_NbReconnectAttempts

Reply:
Oh I see. So m_NbReconnectAttempts is the number of times where we try to reconnect to the game. But for each attempt, if we are using lobby, we try to reconnect to lobby k_NbLobbyReconnectAttempts times, before considering the reconnect attempt as having failed and trying again. The wording is a bit confusing though, I'm open to suggestions to make this clearer.

Reply:
So let me get this straight. A disconnect from lobby happens when the heartbeats to lobby stop. Once the clinet comes back, it can call the reconnect api. There's multiple failure situations here:
client internet is down:
- client will try to call reconnect to lobby multiple times, this will fail. service side, this should timeout the relay connection
host is down:
- lobby reconnect passes with no retry
- will attempt multiple time to connect through UTP/Relay, this will fail.
Everything is slow and unstable:
- could reconnect through lobby after 4/5 tries
- could reconnect to relay/utp after 2/3 tries
For this last case, any reason why we're not just counting the number of tries as a whole connection flow? "You have 3 chances to do the whole flow including lobby and utp/relay, else you timeout"? Why the two steps with different retries? Having a single flow would work as well for the above two scenarios too no?

Reply:
As mentioned in the PR description, the goal was to allow some time between the attempts so that if it is a temporary issue, it could be solved. Otherwise, in your first scenario, we would simply send x number of requests back to back, which would both fail anyways.

For your last scenario, the flow actually is that we have three chances to do the whole reconnect flow. However, for each of those chances, we try to reconnect to lobby multiple times before marking that attempt as having failed and starting a new one, the same way as UTP which tries to connect to the relay server multiple times before marking it as a failed attempt when we call StartClient().

Another simpler solution though could be to only try to reconnect to lobby once per attempt, but adding a cooldown between each attempt. The only thing I'm not sure about with this other solution is for UX, when to update the number of attempts on the popup, after the failed attempt and before the cooldown, or after the cooldown and before the next attempt?

Reply:
+1 on single reconnect per attempt
regarding cooldown, that's actually good to have a cooldown on relay/utp as well.
I'd also link to this page on exponential backoff
https://en.wikipedia.org/wiki/Exponential_backoff
I don't think we have time to implement this ourselves, but you could add a note that this implementation could be up to users.
Regarding UX, to me cooldown is part of the connection time as a player. So update after cooldown? Before next attempt?
It'd also be great to have some of this discussion summarized in comments, those are good things to think about for users.

Reply:
Done, and updated the PR description accordingly